etl/steps/data/garden/demography/2023-04-14/population_density.py

```diff
@@ -0,0 +1,135 @@
+"""Build population density OMM dataset.
+
+This dataset is built using our population OMM dataset and the land area given by FAOSTAT (RL):
+
+    `population_density = population / land_area`
+"""
+
+import pandas as pd
+from owid.catalog import Dataset, DatasetMeta, Table, VariableMeta
+from structlog import get_logger
+
+from etl.helpers import PathFinder, create_dataset
+
+log = get_logger()
+
+# Get paths and naming conventions for current step.
+paths = PathFinder(__file__)
+
+
+def run(dest_dir: str) -> None:
+    log.info("population_density: start")
+
+    #
+    # Load inputs.
+    #
+    # Load dependency datasets.
+    ds_population: Dataset = paths.load_dependency("population")
+    ds_land_area: Dataset = paths.load_dependency("faostat_rl")
+
+    # Read relevant tables
+    tb_population = ds_population["population"]
+    tb_land_area = ds_land_area["faostat_rl_flat"]
+
+    #
+    # Process data.
+    #
+    tb = make_table(tb_population, tb_land_area)
+
+    #
+    # Save outputs.
+    #
+    # Create a new garden dataset with the same metadata as the meadow dataset.
+    ds_garden = create_dataset(
+        dest_dir,
+        tables=[tb],
+        default_metadata=build_metadata(ds_population, ds_land_area),
+    )
+    # Save changes in the new garden dataset.
+    ds_garden.save()
+
+    log.info("population_density: end")
+
+
+def make_table(tb_population: Table, tb_land_area: Table) -> Table:
+    """Create a table with population density data."""
+    # Dataframe population
+    df_population = pd.DataFrame(tb_population).reset_index()
+    # Dataframe land area
+    log.info("population_density: process land area datafame")
+    column_area = "land_area__00006601__area__005110__hectares"
+    df_land_area = (
+        pd.DataFrame(tb_land_area)[[column_area]]
+        .reset_index()
+        .rename(columns={column_area: "area"})
+        .sort_values(["country", "year"])
```

Comment:
Maybe it's worth adding a comment explicitly explaining that you only care about the current area of the country (e.g. Russia's land area changes over time).

Reply:
Good point ー this is actually up for discussion. Thing is, land area is only available since 1961 for some countries, and later for others.

I think this is fine for now, as the population metric considers current borders.